src/gen/sanitize_int.rs

```diff
@@ -0,0 +1,273 @@
+use super::rangeset::RangeSet;
+use super::validation_error::{atom, ErrorCode, ValidationErr};
+use crate::allocator::{Allocator, AtomBuf, NodePtr, SExp};
+use crate::run_program::STRICT_MODE;
+
+fn count_zeros_impl(buf: &[u8]) -> usize {
+    let mut ret: usize = 0;
+    while ret < buf.len() && buf[ret] == 0 {
+        ret += 1;
+    }
+    ret as usize
+}
+
+// this *just* counts zeros. Note that not all zeros are redundant. the last
+// zero byte may be necessary. This has to be fixed up by the caller
+fn count_zeros(a: &Allocator, range_cache: &mut RangeSet, atom_range: &AtomBuf) -> usize {
```

Comment:
Fun (but probably not that useful) discussion of `align_to` here: https://stackoverflow.com/questions/65367552/checking-a-vecu8-to-see-if-its-all-zero